src/sink.cpp

```diff
@@ -83,47 +83,51 @@ static void init_context_info_impl(nlohmann::json& dst,
             throw std::runtime_error(fmt::format("Wrong context_info destination type {}", dst.type_name()));
     }
 
-    for (auto const& itr : context_info)
+    for (auto const& context_info : {log.context_info(), context_info, global_context_info})
```

Comment:
Wouldn't this just create a copy of the 3 `ContextInfo` instances?

Reply:
why?
all are by reference

Reply:
I guess you're right.